datumaro/plugins/open_images_format.py

```diff
@@ -238,6 +289,53 @@ def _load_labels(self, items_by_id):
                     item.annotations.append(Label(
                         label=label_index, attributes={'score': confidence}))
 
+    def _load_bboxes(self, items_by_id):
+        label_categories = self._categories[AnnotationType.label]
+
+        for bbox_path in self._glob_annotations(
+            '*' + OpenImagesPath.BBOX_DESCRIPTION_FILE_SUFFIX
```

Comment:
Probably, we don't need '*' here and can only look for the subsets we loaded earlier. This also should resolve the problem with patching, because each subset file should contain annotations only from that subset (cases, when this is not true can produce a warning, but it should not change reading and patching behaviour).

Reply:
I don't agree. It seems like it would require a lot of hardcoding (which subsets have which filenames for each annotation type) for no clear benefit. Furthermore, this wouldn't work with custom datasets with custom subset names, since we can't predict what file names will be used for those.

But speaking of patching, I think I have found a good way to implement it. What we need is to remember the set of annotation files that we wrote, and then remove all annotation files that are not in that set, but that match one of the annotation file patterns. This is fairly simple to implement, and it's safe, because if a file matches an annotation file pattern, but we haven't created it, then that means that it must describe a subset that disappeared (or perhaps an annotation type that disappeared).

Reply:
Ok, let's put it to another PR and try to find solution later.

Reply:
Yep, working on it.